IPython/core/completer.py

```diff
@@ -651,13 +655,26 @@ def __init__(self, shell=None, namespace=None, global_namespace=None,
         #use this if positional argument name is also needed
         #= re.compile(r'[\s|\[]*(\w+)(?:\s*=?\s*.*)')
 
+        # for the time being, remove our python_completions from the list of
+        # completions and see if dev-user complain that some completion is missing. 
+        # still leave the ability to force enable it though configuration. 
+        # we should remove that on next Stable (5.0)
+        import IPython
+        if '.dev' in IPython.version_info and not self.force_python_matches:
+            self.matchers = []
+        else :
+            self.matchers = [self.python_matches] 
```

Comment:
In general, I don't like code checking its own version number, because it
sets up surprises when you change the version. I think we should just turn
off the old completer and see how it goes. If we spot problems, we can
reenable it before release.